src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -77,8 +94,10 @@ public void StartFileChangeListeningAsync()
                 return newCookie;
             }, LazyThreadSafetyMode.ExecutionAndPublication);
 
-            // file change service is free-threaded. start running it in background right away
-            Task.Run(() => _fileChangeCookie.Value, CancellationToken.None);
+            lock (s_lastBackgroundTaskGate)
+            {
+                s_lastBackgroundTask = s_lastBackgroundTask.ContinueWith(_ => _fileChangeCookie.Value, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
I saw a small amount of overhead related to the lock jumping between threads, which could possibly be mitigated by using `ExecuteSynchronously`. Before doing that, I would want to verify that there is no way `ContinueWith` will execute synchronously on its caller for the case where `s_lastBackgroundTask` is already complete at the time `ContinueWith` is called (which happens to be the main thread here).

With that said, I have a much bigger concern related to the performance of the underlying storage. In the previous implementation, if a storage device is responding slowly then the recovery performance of the algorithm was substantially improved (though far from ideal since if it burned all thread pool threads). The new implementation won't block the thread pool directly, but it could cause file trackers to stop getting registered and `EnsureSubscription()` to manifest as a deadlock from the user's perspective.